### DISCUSSION TITLE
Ok now, assuming that this won't be abandoned and this will get development...

### DISCUSSION THREAD

**Discussion by alidan** [gioxx/MarvellousSuspender#14] [General]
There are a few things that I think we need

1) some form of a feature lock/no update version, something easy for someone to import into chrome and use it outside of chrome store. I could honestly care less about the malicious nature of the great suspender but the complete and abrupt removal of it had me loose around 500~ tabs, but though the use of tabs outliner I can recover nearly everything. 

2) we need a way to parse the great suspender tabs as the marvellous suspender tabs, now i may be an outlier who uses tabs outliner and the great suspender, but I have currently around 70k tabs saved in it many suspended, and re opening them is going to be an abject nightmare. anything to make this less painful than need be would be appreciated. 

3) a no auto update policy if google allows it, each new version could be a numerical increase, as in the marvelous suspender, the marvelous suspender 2, the marvelous suspender 3 ect ect with a user notification that the version we are on is old, get the new one. 

this should solve every issue I have, likely others have, and allow for future versions to not do us raw while we are sleeping.

I would even suggest a secondary extension to just unsuspend tabs if/when chrome decides to yeet it again, because poor planning on the old devs part kind of makes any form of reopening tabs a pain in the ass.

**Comment by cowbert** [comment]
> There are a few things that I think we need
> 
> 1. some form of a feature lock/no update version, something easy for someone to import into chrome and use it outside of chrome store. I could honestly care less about the malicious nature of the great suspender but the complete and abrupt removal of it had me loose around 500~ tabs, but though the use of tabs outliner I can recover nearly everything.

You can already do this yourself. ~~Git checkout~~ download from github either a tag or a commit. Put your Chrome into developer mode and import the `src` directory of your download. Now you have version locked to that tag/commit.

> 2. we need a way to parse the great suspender tabs as the marvellous suspender tabs, now i may be an outlier who uses tabs outliner and the great suspender, but I have currently around 70k tabs saved in it many suspended, and re opening them is going to be an abject nightmare. anything to make this less painful than need be would be appreciated.

This has been addressed many times. In your history you search for the magic string `chrome-extension://klbibkeccnjlkjkiokjodocebajanakg/suspended.html` , then parse the string as a URI. From here, parse the HTTP parameter `uri`. The original URI is the value of said`uri` parameter. It's like 10 lines or fewer of JS.

**Comment by alidan** [comment]
I have absolutely no idea what anything below 2. is at all. the moment that anything script wise is needed im at a near complete loss and everything I have seen so far has required far more than I understand. Are any of the ways to scrub the urls extensions or do they all require python scripting? 

I get understand everything between
chrome-extension://klbibkeccnjlkjkiokjodocebajanakg/suspended.html
and
uri=
gets removed, but I have 0 idea how to do that and notepad++ I dont think will do that.

**Comment by shawnkhall** [comment]
You don't actually need to remove everything between the `chrome-extension` and `&uri=`, but rather just remove `&uri=` and everything before that. The simplest method is to just replace `&uri=` with a carriage return & line feed. Notepad++ will absolutely do that, and it's as easy as CTRL+H (replace), Find: `&uri=`, Replace: `\r\n`, SearchMode: `Extended`, then click `Replace All`. It should take you all of seconds after gathering the URLs, which you can do manually (yuck) or with a [history parsing extension](https://chrome.google.com/webstore/detail/export-historybookmarks-t/dcoegfodcnjofhjfbhegcgjgapeichlf?hl=en).

**Reply by alidan** [reply]
Didn't think of adding a space to notepad++, I know I had to do that in the passed, what I was looking for was a clean input X output Y so I could easily copy paste the results. to that end I found this site 

http://www.molbiotools.com/textextractor.html

and with "uri=" in the followed by field, it removes everything before that on the line, making a very clean way to import text.

I think this is probably the easiest way for people who know what they want but don't know how to get it to do this.

**Comment by TheBoojah** [comment]
> 2\. we need a way to parse the great suspender tabs as the marvellous suspender tabs, now i may be an outlier who uses tabs outliner and the great suspender, but I have currently around 70k tabs saved in it many suspended, and re opening them is going to be an abject nightmare. anything to make this less painful than need be would be appreciated.

This would be amazing! I also use tabs outliner, restoring windows and such from its backup opens the old TGS tabs, where you can indeed manually remove TGS stuff from the url and restore the tab. No idea on how to do it on the tabs outliner database though. You would have to export the database to html and then batch process it and then import it back to tabs outliner.
Perhaps this conversion should be done by tabs outliner itself, but it makes more sense to me to deal with the old TGS sins using the replacement TMS.

**Reply by alidan** [reply]
this is not the first time TGS now TMS has had issues with parsing.

now, from what I gather everything before "uri=" is 'garbage' data, it's presumably there for a reason, my best guess looking at other suspending extensions, its there for force tab closed regardless of if im on the tab or not, and allow chrome to be shut down and reopened without opening in some cases 1000+ tabs

I think that if the extension can suspend every tab one by one, I think it could probably with relative ease parse everything before "uri=" is garbage, and remove it, allowing the recovery of tabs, or in a more ideal solution, do the above, but add in whatever it needs to keep the tab suspended, I have several sites that are 'ping us to many times and we block you for a few days' 

Ideally, adding this function allows the extension to go forward would solve issues that crop up, passed issues that will be a pain for many users, and a change in how the suspended tabs suspend could be a precursor change that signals time to get off the extension.
